pkg/ipamd/ipamd.go

```diff
@@ -412,8 +422,30 @@ func New(rawK8SClient client.Client, cachedK8SClient client.Client) (*IPAMContex
 	checkpointer := datastore.NewJSONFile(dsBackingStorePath())
 	c.dataStore = datastore.NewDataStore(log, checkpointer, c.enablePrefixDelegation)
 
+	clientSet, err := k8sapi.GetKubeClientSet()
+	if err != nil {
+		log.Errorf("Failed to get client set", err)
+		return nil, err
+	}
+
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartRecordingToSink(&typedcorev1.EventSinkImpl{
+		Interface: clientSet.CoreV1().Events(""),
+	})
+	c.eventRecorder = eventBroadcaster.NewRecorder(clientgoscheme.Scheme, corev1.EventSource{
+		Component: "aws-node",
+		Host:      c.myNodeName,
+	})
+
 	err = c.nodeInit()
 	if err != nil {
+		log.Debugf("Failed to initialize node, err: %v", err)
+		var awsErr awserr.Error
+		if errors.As(err, &awsErr) {
```

Comment:
I think we should move this logic to another function. The function can take in "err". Initially do sanity checks and then we can have case statements for each error code we need to broadcast. In that way adding new events will be easy, just need to call the function and add a new case inside the function.

Reply:
Sounds good, will update the PR with these changes.